spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/ErrorParser.java

```diff
@@ -0,0 +1,20 @@
+package org.springframework.cloud.sleuth;
+
+/**
+ * Contract for hooking into process of adding error response tags.
+ * This interface is only called when an exception is thrown upon receiving a response.
+ * (e.g. a response of 500 may not be an exception).
+ *
+ * @author Marcin Grzejszczak
+ * @since 1.2.1
+ */
+public interface ErrorParser {
+
+	/**
+	 * Allows setting of tags when an exception was thrown when the response was received.
+	 *
+	 * @param span - current span in context
+	 * @param error - error that was thrown upon receiving a response
+	 */
+	void parseErrorTags(Span span, Throwable error);
```

Comment:
I think mentioned a while back, but Span allows many more things than setting tags. One way to address this is in docs. For example, you could mention explicitly that implementations should only call span.tag and not any other methods. I'm ok on this regardless.